test/benchmarks/ml/vllm_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2020 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package ml
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types/mount"
+	"gvisor.dev/gvisor/pkg/test/dockerutil"
+	"gvisor.dev/gvisor/test/benchmarks/harness"
+)
+
+func TestMain(m *testing.M) {
+	harness.Init()
+	harness.SetFixedBenchmarks()
+	os.Exit(m.Run())
+}
+
+// BenchmarkVLLM runs a vLLM workload.
+func BenchmarkVLLM(b *testing.B) {
+	doVLLMTest(b)
+}
+
+func doVLLMTest(b *testing.B) {
+	serverMachine, err := harness.GetMachine()
+	if err != nil {
+		b.Fatalf("failed to get machine: %v", err)
+	}
+	defer serverMachine.CleanUp()
+
+	b.Run("opt-125", func(b *testing.B) {
+		ctx := context.Background()
+
+		b.ResetTimer()
+		b.StopTimer()
+
+		for i := 0; i < b.N; i++ {
```

Comment:
The use of `harness.SetFixedBenchmarks` earlier means that this benchmark will run exactly once and `b.N` will be set to `1`. So there is no need for this loop here.

It would be nice to make the benchmark non-fixed, but then that requires the command below needs to be proportional to it in some manner (e.g only do `b.N` requests in the client, for example by only testing a sample of the dataset). If that's possible, then keep this loop here and change the client command to do that. If not, the loop should go away (and can be replaced by a check that `b.N` is indeed 1).

Reply:
b.N is now used as argument for [--num-prompts](https://github.com/vllm-project/vllm/blob/114d7270ffc2e5a66e0974b0d6d913c7f990afa7/benchmarks/benchmark_serving.py#L635), i.e. the number of prompts a client sends per iteration. 

As part of this change I also moved the model setup out of the for loop. This saves a lot of work for loading and unloading the model. I _think_ this is fine as there shouldn't be any state within the serverCtr, but am not entirely certain. I am still experimenting whether I can see differences in the reported metrics between having the serverCtr recreated each iteration and keeping it.
Thoughts?